frontend/src/app/auth/services/auth.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { tap } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthService {
+  constructor(private http: HttpClient) {}
+
+  login(credentials: { email: string; password: string }) {
+    console.log('login', credentials);
+    return this.http
+      .post('http://localhost:8000/auth/login', credentials)
+      .pipe(tap((res: any) => this.setSession(res)));
+  }
+
+  healthCheck() {
+    return this.http.post('http://localhost:8000/auth/health', {
+      test: 'test',
+    });
+  }
+
+  isLoggedIn(): boolean {
+    return new Date().getTime() < Number(localStorage.getItem('expires_at'));
```

Comment:
I'd recommend not to store expires_at in FE. BE already handles expired tokens and returns 401 so you can handle 401 in FE